odoo/orm/fields_relational.py

```diff
@@ -928,6 +1082,8 @@ def __init__(self, comodel_name: str | Sentinel = SENTINEL, relation: str | Sent
             string=string,
             **kwargs
         )
+        if 'auto_join' in kwargs:
+            raise NotImplementedError(f"auto_join is not supported on Many2many fields: {self}")
```

Comment:
This should be in a separate PR.

Reply:
#189430

Reply:
This should be removed, as that PR was merged.